tokio/src/loom/sync/expect_poison.rs

```diff
@@ -0,0 +1,29 @@
+use super::MutexGuard;
+
+/// Extension trait for `MutexGuard` related types (libstd, parking_lot, etc.),
+/// providing consistent handling of Mutex poisoning.
+pub(crate) trait ExpectPoison<T> {
```

Comment:
What about, instead of introducing a trait, we define our own `Mutex` type that is a wrapper around either `std::Mutex` or `parking_lot::Mutex`. The implementation would be conditionally defined based on the feature flag?

Reply:
What's the advantage, or what don't you like about the way it is currently, with the extension trait?  To me this seems very minimal and zero cost. What's the perceived downside?

Reply:
I think the easiest strategy is to just use the std::Mutex api, and when using parking_lot, define a wrapper around the parking_lot mutex that returns Result